### ISSUE TITLE
Multiscale Image Rendering

### ISSUE THREAD

**Issue by tathey1** [napari/napari#3556]
## 📚 Documentation
I am trying to use napari to view spatially large 3d images, so I am exploring the pyramid format options. I decided to start by looking at a 3d image after a pyramid_gaussian operation (inspired by [this example script](https://github.com/napari/napari/blob/main/examples/add_multiscale_image.py)). I was expecting the viewer to dynamically load certain resolutions according the the viewer's zoom level. However, I noticed that the pyramid image looks significantly different than the original image.

Below I am attaching:
- The code where I display both the full pyramid, and only the highest resolution of the pyramid
- The napari viewer  (in 2d mode) where i show that the different images look different

```
from skimage.transform import pyramid_gaussian
multiscale = list(pyramid_gaussian(im2, downscale=2, max_layer=4))

viewer = napari.Viewer(ndisplay=2)
viewer.add_image(multiscale, rgb=False, multiscale=True, name="multiscale image", scale=[0.3,0.3,1])
viewer.add_image(multiscale[0], rgb=False, name="image", scale=[0.3,0.3,1])
```

![im](https://user-images.githubusercontent.com/18537350/139307729-b1860900-a1be-4943-83c7-cfb8776c6711.png)

![multi](https://user-images.githubusercontent.com/18537350/139307754-0a237491-0a9a-4a90-b63c-e8448220d585.png)

It looks like a contrast issue, but I tried different contrast levels and could never get the images to match. I could not find any details in the documentation why this might be the case.

**Comment by jni** [comment]
Hey @tathey1!

There's a few different things that could be going on here. One thing is that in 2D view we *are* loading the corresponding level of the pyramid, but note that you are downscaling along all axes, so the low resolution level that you are viewing contains "data" from adjacent slices in the full resolution volume.

*Also*, note that scikit-image [pyramid_gaussian](https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.pyramid_gaussian) implicitly converts your images from uint8 to float, so that might mess with napari's autocontrast calculation — does the issue get "fixed" (again, they will never exactly match because you are "blending" adjacent slices when using downscaling) by clicking on "autoscale: once" for the pyramid layer?

Finally, I will note that *currently* napari does *not* do any dynamic loading when viewing a multiscale dataset in 3D view: it only loads the lowest resolution image. But that's not the issue here, since you are viewing the data in 2D slices.

**Comment by tathey1** [comment]
Thanks for the response @jni. After playing around a little according to your suggestions, the solution that gave me the functionality I wanted was enabling the "continuous" autoscaling. Then, when I zoomed in on the pyramid layer, the image darkened and things seemed to match.

Before I close this issue though, I want to ask, does the team have plans to do dynamic loading in 3d view? It's something that would be very useful in my neuron reconstruction work, so I would consider trying to contribute that feature.

**Comment by GenevieveBuckley** [comment]
@tathey1 yes, there is some work going on around better rendering of multi-resolution datasets in 3D. If you'd like to join/contribute to any of it, I'm sure that would be very welcome.

You can see a recent discussion about this here: https://napari.zulipchat.com/#narrow/stream/212875-general/topic/poor-mans-octree

It's not at the stage where there is a pull request in napari I can point you to. Instead, look at this PR https://github.com/alisterburt/napari/pull/9 for the "poor-mans-octree" branch in Alister's fork of napari, where we have been playing with two example scripts named `are_the_points_in_view.py` and `are_the_chunks_in_view.py`.

Juan has [described the next steps in this comment](https://napari.zulipchat.com/#narrow/stream/212875-general/topic/poor-mans-octree/near/321985873). Feel free to comment on that discussion thread if you'd like to get involved.
